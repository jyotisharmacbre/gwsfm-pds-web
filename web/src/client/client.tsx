import axios from 'axios';
import authentication from '@kdpw/msal-b2c-react';
import appConfig from '../helpers/config-helper';

const config = appConfig();
export const baseURL = config.REACT_APP_MIDDLETIER_URL;
export const baseAPI = axios.create({ baseURL });
export const userServiceURL = config.REACT_APP_GATEWAY_URL;
export const userServiceAPI = axios.create({ baseURL: userServiceURL });

const isTokenHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('tokenHandlerEnabled') &&
    !config['tokenHandlerEnabled']
    ? false
    : true;
};

const isErrorHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('errorHandlerEnabled') &&
    !config['errorHandlerEnabled']
    ? false
    : true;
};

const isSuccessHandlerEnabled = (config = {}) => {
  return config.hasOwnProperty('successHandlerEnabled') &&
    !config['successHandlerEnabled']
    ? false
    : true;
};

const requestHandler = request => {
  if (isTokenHandlerEnabled(request)) {
    // Add request token here
    const token = authentication.getAccessToken();
    request.headers['accessToken'] = token;
  }
  return request;
};

baseAPI.interceptors.request.use(request => requestHandler(request));
userServiceAPI.interceptors.request.use(request => requestHandler(request));

const errorHandler = error => {
  if (isErrorHandlerEnabled(error.config)) {
    // Handle errors
  }
  return Promise.reject({ ...error });
};

const successHandler = response => {
  if (isSuccessHandlerEnabled(response.config)) {
    // Handle responses
  }
  return response;
};

baseAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);

userServiceAPI.interceptors.response.use(
  response => successHandler(response),
  error => errorHandler(error)
);
