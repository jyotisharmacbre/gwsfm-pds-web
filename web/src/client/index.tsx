export { baseAPI, userServiceAPI} from './client';
