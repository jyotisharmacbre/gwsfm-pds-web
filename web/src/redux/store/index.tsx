import { applyMiddleware, createStore } from 'redux';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunk from 'redux-thunk';

import allReducer from '../reducers';

const persistConfig = {
  key: 'root',
  storage
};

const persistedReducer = persistReducer(persistConfig, allReducer);

const middleware = applyMiddleware(thunk);
const store = createStore(persistedReducer, middleware);
const persistor = persistStore(store);
export { store, persistor };
