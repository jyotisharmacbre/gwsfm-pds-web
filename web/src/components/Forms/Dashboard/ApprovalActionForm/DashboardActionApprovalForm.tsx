import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { IState } from '../../../../store/state';
import moment from 'moment';
import { Link } from 'react-router-dom';
import GridTable from '../../../Table/GridTable';
import { injectIntl, FormattedMessage } from 'react-intl';
import { reduxForm, InjectedFormProps } from 'redux-form';
import { IProjectDashboardGrid } from '../../../../store/Dashboard/Types/IProjectDashboardGrid';
import IReactIntl from '../../../../Translations/IReactIntl';
import { ILookup } from '../../../../store/Lookups/Types/ILookup';
import { getLookupDescription, displayUserName } from '../../../../helpers/utility-helper';
import { LookupItems } from '../../../../helpers/constants';
import { IUserServiceData } from '../../../../store/UserService/Types/IUserService';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExclamation } from '@fortawesome/free-solid-svg-icons';
import { formatMessage } from '../../../../Translations/connectedIntlProvider';
import useConfigContext from '../../../../hooks/useConfigContext';
import DashboardApprovalStatusAndClassMapping from '../../../../store/Dashboard/Types/DashboardApprovalStatusAndClassMapping';

interface Props {
	actionApprovalValues: Array<IProjectDashboardGrid>;
	showValues: number;
	lookupValues: Array<ILookup>;
	userNamesForEmailsValues: Array<IUserServiceData>;
}
let DashboardActionApprovalForm: React.FC<Props & InjectedFormProps<Array<IProjectDashboardGrid>, Props>> = (props) => {
	const [gridData, setGridData] = useState<Array<IProjectDashboardGrid>>([]);
	const config = useConfigContext();
	useEffect(
		() => {
			if (
				props.actionApprovalValues &&
				props.actionApprovalValues.length > 0 &&
				props.lookupValues &&
				props.userNamesForEmailsValues &&
				props.showValues
			) {
				const actionApprovalvalues = JSON.parse(JSON.stringify(props.actionApprovalValues));
				setGridData(
					getActionApprovalValues(
						actionApprovalvalues,
						props.lookupValues,
						props.userNamesForEmailsValues,
						props.showValues
					)
				);
			}
		},
		[props.actionApprovalValues, props.lookupValues, props.userNamesForEmailsValues, props.showValues]
	);

	const getActionApprovalValues = (data, allLookups, namesAndEmails, countVals) => {
		let result =
			data &&
			data.map(function (rowProject) {
				if (allLookups.length > 0 && namesAndEmails) {
					let approvalStatusClass = DashboardApprovalStatusAndClassMapping.find(x => x.approvalStatus == rowProject.approvalStatus);
					rowProject.approvalStatus = !isNaN(rowProject.approvalStatus) ? getLookupDescription(
						allLookups,
						rowProject.approvalStatus,
						LookupItems.Project_Approval_Sign_Off_Status
					) : rowProject.approvalStatus;
					rowProject['approvalStatusClass'] = `status_design ${approvalStatusClass?.className}`
					var mailObj = namesAndEmails.find(
						lk => lk.email && rowProject.modifiedBy && lk.email.toUpperCase() === rowProject.modifiedBy.toUpperCase()
					);
					rowProject.modifiedBy = mailObj
						? `${displayUserName(mailObj)}`
						: rowProject.modifiedBy;
					rowProject.name = (
						<Link
							to={{
								pathname: `/ReviewApprove/${rowProject.projectId}`,
								state: { fromDashboard: true }
							}}
						>
							{rowProject.name}
						</Link>
					);
					rowProject.modifiedOn =
						rowProject.modifiedOn != '' ? moment(rowProject.modifiedOn).format(config.REACT_APP_DATE_FORMAT) : '';
				}
				return rowProject;
			});
		return result;
	};
	return (
		<React.Fragment>
			{gridData.length > 0 ? (
				<GridTable
					columns={getTableColumns()}
					data={gridData}
					sorting={false}
					className="price-table home_screen_table"
					ActionList={[]}
				/>
			) : (
					<span className="table-row-no-pending-actions">
						<FontAwesomeIcon className="active" icon={faExclamation} />
						<FormattedMessage id="HOMESCREEN_GRID_NO_ACTION_MESSAGE" />
					</span>
				)}
		</React.Fragment>
	);
};
const getTableColumns = () => {
	return [
		{
			title: formatMessage("HOMESCREEN_GRID_COLUMN_NAME"),
			field: 'name'
		},
		{
			title: formatMessage("HOMESCREEN_GRID_COLUMN_UPDATEDBY"),
			field: 'modifiedBy'
		},
		{
			title: formatMessage("HOMESCREEN_GRID_COLUMN_DATE"),
			field: 'modifiedOn'
		},
		{
			title: formatMessage("HOMESCREEN_GRID_COLUMN_STATUS"),
			field: 'approvalStatus',
			class: 'approvalStatusClass'
		}
	];
};

const mapStateToProps = (state: IState) => ({
	initialValues: state.dashboardGrid.actionApprovalDetails
});

const form = reduxForm<Array<IProjectDashboardGrid>, Props>({
	form: 'DashboardActionApprovalForm',
	enableReinitialize: true
})(DashboardActionApprovalForm);
export default connect(mapStateToProps)(form);
