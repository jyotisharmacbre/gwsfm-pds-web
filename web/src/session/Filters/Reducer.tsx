import { IFilterState } from '../state';
import { Reducer } from 'redux';

const initialFiltersState: IFilterState = {
  projectname: '',
  customer_client: '',
  owner: '',
  contracttype: '',
  probofwinning: undefined,
  status: '',
  cdmnotifiable: false,
  soldmargin: undefined,
  expectedstartdate: undefined,
  approximatevalue: undefined,
  weightedtcv: undefined,
  validForm: false
};

const filterReducer: Reducer<IFilterState> = (
  state = initialFiltersState,
  action
) => {
  switch (action.type) {
    case 'GetFilterAction': {
      return {
        ...state
      };
    }
    case 'FilterAddAction': {
      return Object.assign({}, state, action.form);
    }

    default:
      return state;
  }
};

export default filterReducer;
