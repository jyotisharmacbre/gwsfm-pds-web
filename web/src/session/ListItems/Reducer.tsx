import { ListItemActions } from "./Type";
import { IListState } from "../state";
import { Reducer } from "redux";

const initialState: IListState = {
    locales: [{ name: '--Please choose an option---', value: '' }],
    customerContract: [{ name: '--Please choose an option---', value: '' }],
    error: ''
};

const listItemReducer: Reducer<IListState, ListItemActions> = (
    state = initialState, action,
) => {
    switch (action.type) {
        case 'GetLocalesBegin':
            return { ...state }
        case 'GetLocalesSuccess':
            return Object.assign({}, state, { locales: action.data });
        case 'GetLocalesFailure':
            return { ...state, error: action.error };
        case 'GetCustomerContractBegin':
            return { ...state }
        case 'GetCustomerContractSuccess':
            return Object.assign({}, state, { customerContract: action.data });
        case 'GetCustomerContractFailure':
            return { ...state, error: action.error };
        default:
            return state
    }
}

export default listItemReducer;