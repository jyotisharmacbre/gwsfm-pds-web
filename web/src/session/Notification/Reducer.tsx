import { NotificationActions } from './Type';
import { INotificationState } from '../state';
import { Reducer } from 'redux';

const initialState: INotificationState = {
  notificationCount: 0
};

const notificationReducer: Reducer<INotificationState, NotificationActions> = (
  state = initialState,
  action
) => {
  switch (action.type) {
    case 'GetNotificationBeginAction':
      return { ...state };
    case 'GetNotificationSuccessAction':
      return Object.assign({}, state, action.data);
    //  case 'GetNotificationFailureAction':

    default:
      return state;
  }
};

export default notificationReducer;
