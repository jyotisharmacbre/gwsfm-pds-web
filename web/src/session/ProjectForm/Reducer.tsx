import { IProjectFormState } from "../state";
import { ProjectFormActions } from "./Type";
import { Reducer } from "redux";

const initialState: IProjectFormState = {
    company: 'CBRE Managed Services',
    customer_contract: 'RS Electrical',
    pmexperience: true,
    locale: 'English',
    projectscope: '',
    projectmanager: '',
    projectname: '',
    headofproject:'',
    cnnumber:'',
    comments:'',
    typeofengagement:'',
    currency:'',
    projectowner:'',
    probofwinning:0,
    approximatevalue:0,
    contracttype:'',
    cdmnotifiable:false,
    projectstatus:'',
    assetworkedonprimary:'',
    assetworkedonsecond:'',
    assetworkedonthird:'',
    soldmargin:0,
    weightedtcv:0,
    rank:0,
    validForm: false,
    invalidCompany: false,
    invalidCustomerContract: false,
    invalidLocale: false,
    invalidPMExperience: false,
    invalidProjectName: false,
    invalidProjectScope: false,
    invalidProjectManager: false,
    invalidApproxValue: false,
    invalidAssetsWorkedOnPrimary: false,
    invalidCMDNotifiable:false,
    invalidContractType:false,
    invalidCurrency:false,
    invalidProbOfWinning: false,
    invalidProjectOwner:false,
    invalidHeadOfProject: false,
    InvalidProjectOwner: false
};

const projectReducer: Reducer<IProjectFormState, ProjectFormActions> = (
    state = initialState, action,
) => {
    switch (action.type) {
        case 'GetProjectFormAction':
            {
                return {
                    ...state
                }
            }
        case 'ProjectFormAddAction':
            {
                return Object.assign({}, state, action.form);
            }

        default:
            return state
    }
}

export default projectReducer;