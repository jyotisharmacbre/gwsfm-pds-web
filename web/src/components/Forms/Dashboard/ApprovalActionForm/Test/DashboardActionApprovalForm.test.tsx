import React from 'react';
import { mount, ShallowWrapper } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../../../../../store';
import { IntlProvider } from 'react-intl';
import translations from '../../../../../Translations/translation';
import {
  intialDashboardState,
  intialLookupvalues,
  intialUsersEmailsData,
  getDashboardData,
  getUsersEmailData
} from './DashboardActionApprovalFormTestData';
import { reducer as formReducer } from 'redux-form';
import DashboardActionApprovalForm from '../DashboardActionApprovalForm';
import { MemoryRouter } from 'react-router-dom';
import { ActionType } from '../../../../../store/Dashboard/Types/ActionType';
import { initialState } from '../../../../../store/Dashboard/Reducer';
import DashboardGridDetailReducer from '../../../../../store/Dashboard/Reducer';
import nock from 'nock';
import { baseURL, userServiceURL } from '../../../../../client/client';

nock(baseURL)
  .get('/api/users/pendingApprovals')
  .reply(200, getDashboardData);

nock(userServiceURL)
  .post('/api/users/getusernamesforemailids')
  .reply(200, getUsersEmailData);
let props = {
  actionApprovalValues: intialDashboardState.actionApprovalDetails.map(a =>
    Object.assign({}, a)
  ),
  showValues: 5,
  lookupValues: intialLookupvalues,
  userNamesForEmailsValues: intialUsersEmailsData
};
describe('Dashboard Form testCases', () => {
  let wrapper: any;
  const componentMount = props => {
    wrapper = mount(
      <Provider store={store}>
        <IntlProvider locale="en" messages={translations['en'].messages}>
          <MemoryRouter>
            <DashboardActionApprovalForm {...props} />
          </MemoryRouter>
        </IntlProvider>
      </Provider>
    );
  };

  it('Defines the component', () => {
    componentMount({ props });
    expect(wrapper).toBeDefined();
  });
  it('should match the snapshot', () => {
    componentMount({ props });
    expect(wrapper).toMatchSnapshot();
  });
});
// describe('Dashboard form reducer', () => {
//   it('should handle Get PROJECT DASHBOARD GRID DETAILS successfully', () => {
//     const projectDashboardGridAction: any = {
//       type: ActionType.PROJECT_DASHBOARD_GRID_DETAILS
//     };
//     expect(
//       DashboardGridDetailReducer(initialState, projectDashboardGridAction)
//     ).toMatchSnapshot();
//   });
//});
