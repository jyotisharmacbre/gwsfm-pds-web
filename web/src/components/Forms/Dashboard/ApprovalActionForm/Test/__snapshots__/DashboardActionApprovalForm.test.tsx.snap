// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Form testCases should match the snapshot 1`] = `ReactWrapper {}`;

exports[`Dashboard form reducer should handle Get PROJECT DASHBOARD GRID DETAILS successfully 1`] = `
Object {
  "actionApprovalDetails": undefined,
  "error": null,
  "loading": false,
}
`;

exports[`Dashboard form reducer should handle Get RESET DASHBOARD STATE successfully 1`] = `
Object {
  "actionApprovalDetails": Array [],
  "error": null,
}
`;

exports[`Dashboard form reducer should handle PROJECT DASHBOARD GRID ERROR successfully 1`] = `
Object {
  "actionApprovalDetails": Array [],
  "error": undefined,
  "loading": false,
  "notify": 2,
}
`;
